File: acct-catalog-ws-spec/src/main/java/com/desolatetimelines/acct/catalog/ws/model/IconCreateRequest.java
package com.desolatetimelines.acct.catalog.ws.model;

/**
 * Defines the body for an icon creation request
 *
 * @param iconName         The name to be given to the icon (must be unique per category)
 * @param iconCategoryName The name of the icon category the icon is to be mapped to (can be a new category name)
 * @param iconBase64       The base64-encoded bytes of the icon's image file
 */
public record IconCreateRequest(
    String iconName,
    String iconCategoryName,
    String iconBase64
) {
}

File: acct-gui-angular/src/app/model-acct/icon-create-request.ts
/**
 * Defines the body for an icon creation request
 */
export interface IconCreateRequest {

    /**
     * The name to be given to the icon (must be unique per category)
     */
    iconName : string

    /**
     * The name of the icon category the icon is to be mapped to (can be a new category name)
     */
    iconCategoryName : string

    /**
     * The mime type of the icon file
     */
    iconMimeType : string

    /**
     * The base64-encoded bytes of the icon's image file
     */
    iconBase64 : string

}